tools/tertiary-analysis/pyscenic/pyscenic_aucell.xml

```diff
@@ -0,0 +1,110 @@
+<tool id="pyscenic_aucell" name="PySCENIC AUCell" profile="21.09" version="0.12.1+galaxy0">
+    <description>calculates AUCell to find relevant regulons/gene sets</description>
+    <requirements>
+        <container type="docker">
+            aertslab/pyscenic:0.12.1
+        </container>
+    </requirements>
+    <command><![CDATA[
+        ln -s "${expression_mtx_fname}" expr_mat.loom &&
+        ln -s "${signatures_fname}" regulons.tsv &&
+
+        pyscenic aucell expr_mat.loom regulons.tsv
+           -o aucell.tsv
+        #if $transpose
+           -t
+        #end if
+        #if $weights
+           -w
+        #end if
+           --num_workers \${GALAXY_SLOTS:-1}
+        #if $seed
+           --seed "${seed}"
+        #end if
+        #if $rank_threshold
+           --rank_threshold "${rank_threshold}"
+        #end if
+        #if $auc_threshold
+           --auc_threshold "${auc_threshold}"
+        #end if
+        #if $nes_threshold
+           --nes_threshold "${nes_threshold}"
+        #end if
+        #if $cell_id_attribute
+           --cell_id_attribute "${cell_id_attribute}"
+        #end if
+        #if $gene_attribute
+           --gene_attribute "${gene_attribute}"
+        #end if
+        #if $sparse
+           --sparse
+        #end if
+
+        && mv aucell.tsv "${output}"
+    ]]></command>
+  
+    <inputs>
+      <param name="expression_mtx_fname" format="loom" type="data" label="Expression Matrix File" help="The file that contains the expression matrix for the single-cell experiment. Supported formats: csv (rows=cells x columns=genes) or loom (rows=genes x columns=cells)." />
+      <param name="signatures_fname" type="data" format="csv,tabular" label="Gene Signatures/Regulons File" help="The file that contains the gene signatures (usually the precomputed regulons). Currently only csv/tsv supported, could be extended." />
+      <param name="transpose" type="boolean" label="Transpose Expression Matrix" help="Transpose the expression matrix if supplied as csv (rows=genes x columns=cells)." optional="true" />
```

Comment:
an optional boolean?

Reply:
I write a lot of these through an LLM, so there will be things that escape like this, thanks for pointing it out!

Reply:
Instead of an LLM you should probably use: https://github.com/galaxyproject/galaxy-language-server

Or  at least in conjunction.

Reply:
:-) please also try the auto-format option. This will help reviewing a lot. It also has macros support.

Reply:
Thanks for the autoformat tip, wasn't aware that Dave module's had it for galaxy tools, very useful! Used it and moved certain things to macros with the refactoring utiity from his VSCode module, very handy as well.